lib/pages/chat_list/chat_list_view_builder.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:fluffychat/pages/chat_list/chat_list.dart';
+import 'package:fluffychat/pages/chat_list/chat_list_item.dart';
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+class ChatListViewBuilder extends StatelessWidget {
+  final ChatListController controller;
+  final List<Room> rooms;
+
+  const ChatListViewBuilder({
+    super.key,
+    required this.controller,
+    required this.rooms,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView.builder(
+      shrinkWrap: true,
+      physics: const NeverScrollableScrollPhysics(),
+      itemCount: rooms.length,
+      itemBuilder: (BuildContext context, int index) {
+        return ValueListenableBuilder(
+          valueListenable: controller.selectModeNotifier,
+          builder: (context, selectedRoomIds, _) {
```

Comment:
i don't see where you use `selectedRoomIds`

Reply:
Yes, it triggers rebuild when `selectMode` change